apis/core/v1alpha2/feature_types.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FeatureSpec defines the desired state of Feature
+type FeatureSpec struct {
+	// Description of the feature.
+	Description string `json:"description,omitempty"`
+	// Immutable indicates this feature cannot be toggled once set
+	// If set at creation time, this state will persist for the life of the cluster
+	Immutable bool `json:"immutable"`
+	// Discoverable indicates if clients should include consider the Feature available for their use
+	// Allowing clients to control discoverability is one of the ways the API allows gradual rollout of functionality
+	Discoverable bool `json:"discoverable"`
+	// Activated defines the default state of the features activation
+	Activated bool `json:"activated"`
+	// Stability indicates stability level of this feature.
+	// Stability levels are Work In Progress, Experimental, Technical Preview, Stable and Deprecated.
+	// +kubebuilder:validation:Enum=Work In Progress;Experimental;Technical Preview;Stable;Deprecated
+	// - Work In Progress: the default for new resources, represents local dev. intended to be hidden and deactivated
+	// - Experimental: the first milestone meant for limited wider consumption, discoverable and defaults to deactivated
+	// - Technical Preview: greater visibility for proven designs, discoverable and defaults to activated
+	// - Stable: intended to be part of the mainline codebase, non-optional
+	// - Deprecated: destined for future removal
+	Stability string `json:"stability"`
+}
+
+// FeatureStatus defines the observed state of Feature
+type FeatureStatus struct{}
+
+// Feature is the Schema for the features API
+// +kubebuilder:object:root=true
```

Comment:
Nit: sometimes in kubebuilder markers there is a space after the `//` comment tags and other times there aren't. I think we should just choose a format and stick with it. I think it's more common not to have that space, so I say remove these spaces, but this is a nit, also, so your choice.

Applies to lines 24 and 52, and probably others, as well.